crates/rethnet_rpc_server/src/hardhat_methods.rs

```diff
@@ -0,0 +1,766 @@
+                                                    "nodeType": "VariableDeclaration",
+                                                    "scope": 8,
+                                                    "src": "88:3:0",
+                                                    "stateVariable": false,
+                                                    "storageLocation": "default",
+                                                    "typeDescriptions": {
+                                                        "typeIdentifier": "t_int256",
+                                                        "typeString": "int256"
+                                                    },
+                                                    "typeName": {
+                                                        "id": 2,
+                                                        "name": "int",
+                                                        "nodeType": "ElementaryTypeName",
+                                                        "src": "88:3:0",
+                                                        "typeDescriptions": {
+                                                            "typeIdentifier": "t_int256",
+                                                            "typeString": "int256"
+                                                        }
+                                                    },
+                                                    "visibility": "internal"
+                                                }
+                                            ],
+                                            "src": "87:5:0"
+                                        },
+                                        "scope": 9,
+                                        "src": "45:61:0",
+                                        "stateMutability": "pure",
+                                        "virtual": false,
+                                        "visibility": "public"
+                                    }
+                                ],
+                                "scope": 10,
+                                "src": "13:107:0"
+                            }
+                        ],
+                        "src": "13:118:0"
+                    },
+                    "id": 0
+                }
+            }
+        }"##;
+
+        let call = HardhatMethodInvocation::AddCompilationResult(
+            String::from("0.8.0"),
+            serde_json::from_str::<add_compilation_result::CompilerInput>(compiler_input_json)
+                .unwrap(),
+            serde_json::from_str::<add_compilation_result::CompilerOutput>(compiler_output_json)
+                .unwrap(),
+        );
+
+        help_test_method_invocation_serde(call.clone());
+
+        match call {
+            HardhatMethodInvocation::AddCompilationResult(_, ref input, ref output) => {
+                assert_eq!(
```

Comment:
Is there a reason you're  no longer using the same helper method to do this - as for the Eth tests?

Reply:
Only that I wasn't sure of the best way to share it, since the `tests` submodules that contain each of these helpers are private. Where would be a good place for such a common helper to live?

Reply:
Make that helper function generic over the method invocation type in order to avoid the circular dependency.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/3f8b866e69434a869e9bdbc78084c445928699b9